### PR TITLE
Add Hausdorff Clustering Notebooks

### LINKED ISSUE #1013: [DOC]: Trajectory Clustering Notebook based on Hausdorff Distance
### How would you describe the priority of this documentation request

Low (would be nice)

### Is this for new documentation, or an update to existing docs?

New

### Describe the incorrect/future/missing documentation

In GTC presentation, one slide mentions example of trajectory clustering based on hausdorff distance. We should introduce this notebook to `notebooks/`.

### If this is a correction, please provide a link to the incorrect documentation. If this is a new documentation request, please link to where you have looked.

_No response_

### PR DESCRIPTION
**Pull request by isVoid**
## Description

This PR refreshes the hausdorff clustering example as a notebook.
closes #1013

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by isVoid** [comment]
Pending update to hausdorff API in #924

**Comment by harrism** [comment]
Is anything blocking getting this notebook into 23.06, @isVoid ?

**Comment by isVoid** [comment]
Need to clear dataset via SWIPAT, also need to figure out if we want to host the dataset ourselves.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by isVoid** [comment]
<p>I believe so, there's a backdrop image for this notebook that I forgot to upload.</p>

---

 <a href='https://app.reviewnb.com/rapidsai/cuspatial/pull/922/discussion/#comment-1508817823'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: dependencies.yaml [hunk 1]
```diff
@@ -170,10 +170,13 @@ dependencies:
     common:
       - output_types: [conda, requirements]
         packages:
+          - opencv
```
Comment (harrism) [inline dependencies.yaml:173]: Are all these new dependencies just for this notebook? Seems like a lot to impose on all users of cuSpatial if it's only for one example notebook.

Comment (isVoid) [reply dependencies.yaml:173]: opencv is only used to visualize the cluster results. We should be able to mimic the behavior via shapely and matplotlib. Or as @jarmak-nv suggests geopandas can do that job.

Comment (harrism) [reply dependencies.yaml:173]: There are a few other added dependencies other than opencv below this.

**Comment by isVoid** [comment]
<p>It does not. Will remove.</p>

---

 <a href='https://app.reviewnb.com/rapidsai/cuspatial/pull/922/discussion/#comment-1512262149'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by isVoid** [comment]
<p>Can't find specific context relating to "wrongway" from the original blog post. The function itself is just transforming the trajectory coordinate from lonlat space to image space.</p>

---

 <a href='https://app.reviewnb.com/rapidsai/cuspatial/pull/922/discussion/#comment-1512262159'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: dependencies.yaml (hunk 1 above)
Comment (isVoid) [reply dependencies.yaml]: All dependencies here are only installed when testing notebooks. They are not dependencies for python/c++ builds.

I replaced opencv with matplotlib plotting but kept the rest. Ipywidget is lightweight and is helpful for manual hyperparameter tuning. Scikit-learn is the lightest weight image I/O package that I can find.

**Comment by isVoid** [comment]
I went forward with keeping the dependencies in `dependencies.yaml` instead of installing them in the notebook. Installing what's missing in my current development environment in the notebook seems like a heavy assumption imposed on what the user will be running the notebook from.

**Review by harrism — APPROVED** [review]

**Comment by isVoid** [comment]
@thomcom I couldn't find a black formatter that works out of the box with jupyter cells. I tried [this](https://github.com/drillan/jupyter-black) and several provided by jupyter extension market. But autopep8 formatter works nicely with vs-code jupyter editor so I went with that.

**Comment by thomcom** [comment]
<p>"the best result"</p>

---

 <a href='https://app.reviewnb.com/rapidsai/cuspatial/pull/922/discussion/#comment-1521878740'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by thomcom — APPROVED** [review]
One missing "the" in the notebook, otherwise I'm clear.

**Pull request merged**